### PR TITLE
removal of RESCRIPt RNA types, etc..

### PR DESCRIPTION
**Pull request by mikerobeson**
As RNA types, formats, and transformers have been added to base QIIME 2 as of PR (qiime2/q2-types#256), this created problems with RESCRIPt's own RNA types. This PR removes all RESCRIPt RNA types and uses the new base QIIME 2 RNA types. Some transformers (*i.e.* RNA -> DNA) in RESCRIPt that should be migrated over to `q2-types`. But I left them alone for now.

All unit tests and flake8 checks pass. I've also installed these PR changes into my `qiime2-dev` environment w/o warnings of duplicated types, etc...

### CODE REVIEW

File: rescript/types/tests/test_types_formats_transformers.py
Comment (misialq) [inline rescript/types/tests/test_types_formats_transformers.py:235]:
```suggestion
class TestDNAIteratorTransformers(RescriptTypesTestPluginBase):
```

**Review by misialq — COMMENTED** [review]
Thanks @mikerobeson, good catch! Looks good - just a little flake issue. As to the transformers: should we maybe at least create an issue for transferring them over to q2-types, not to forget later?

I'll approve once the CI ran through ✅

File: rescript/types/tests/test_types_formats_transformers.py
Comment (mikerobeson) [reply rescript/types/tests/test_types_formats_transformers.py]: I realized I did not commit this particular change. It was all ready to go in my local git branch. 🤦

**Comment by BenKaehler** [comment]
Thank you @mikerobeson and @misialq!

@mikerobeson, could please update the readme to say that the minimum qiime version is now 2021.4?

@misialq, do not merge! until after the 2021.4 release. (I’m sure you knew that already, but best to be explicit.)

**Review by thermokarst — APPROVED** [review]
LGTM!

**Pull request merged**
